DFC.ServiceTaxonomy.ApiFunction/StartUp/FunctionStartupExtension.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using DFC.ServiceTaxonomy.ApiFunction.Helpers;
+using DFC.ServiceTaxonomy.ApiFunction.Models;
+using DFC.ServiceTaxonomy.ApiFunction.StartUp;
+using Microsoft.Azure.Functions.Extensions.DependencyInjection;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+[assembly: FunctionsStartup(typeof(FunctionStartupExtension))]
+
+namespace DFC.ServiceTaxonomy.ApiFunction.StartUp
+{
+    public class FunctionStartupExtension : FunctionsStartup
+    {
+        public override void Configure(IFunctionsHostBuilder builder)
+        {
+
+            builder.Services.AddOptions<ServiceTaxonomyApiSettings>()
+                .Configure<IConfiguration>((settings, configuration) => { configuration.Bind(settings); });
+
+            builder.Services.AddSingleton<IHttpRequestHelper, HttpRequestHelper>();
```

Comment:
Singletons in Azure Functions are [reused across function executions](https://docs.microsoft.com/en-us/azure/azure-functions/functions-dotnet-dependency-injection#service-lifetimes).
`HttpRequestHelper` uses a [`StreamReader`, which is not thread-safe](https://docs.microsoft.com/en-us/dotnet/api/system.io.streamreader?view=netcore-3.0).
That means that we're likely to see intermittent buggy behavior unless we change the lifetime to Scoped or Transient. Currently either would behave the same, but it would be safer to use Transient, in case we later change the threading model.
Ideally, we should have unit tests that test multi-threaded behaviour, if we use any Singletons.